storm/src/main/storm/mesos/schedulers/SchedulerUtils.java

```diff
@@ -31,49 +34,73 @@
 import java.util.List;
 import java.util.Map;
 
+import static storm.mesos.resources.ResourceEntries.RangeResourceEntry;
+import static storm.mesos.resources.ResourceEntries.ScalarResourceEntry;
+
 public class SchedulerUtils {
 
   private static final Logger log = LoggerFactory.getLogger(SchedulerUtils.class);
 
-  public static boolean isFit(Map mesosStormConf, OfferResources offerResources, TopologyDetails topologyDetails, boolean supervisorExists) {
+  public static List<RangeResourceEntry> getPorts(OfferResources offerResources, int requiredCount) {
+    List<RangeResourceEntry> retVal = new ArrayList<>();
+    List<RangeResourceEntry> resourceEntryList = offerResources.getAllAvailableResources(ResourceType.PORTS);
+
+    for (RangeResourceEntry rangeResourceEntry : resourceEntryList) {
+      Long begin = rangeResourceEntry.getBegin();
+      Long end = rangeResourceEntry.getEnd();
+      for (int i = 0; i <= (end - begin) && requiredCount > 0; i++) {
```

Comment:
This is a bit confusing... Let's look at an example: `begin = 5`, `end = 10`, `requiredCount = 3` 
In this case, the for loop is basically `for (i = 0, i < 5 && requiredCount > 0; i++)`, which would mean that the return for this would be a list of RangeResourceEntries (RRE's) as follows: `[RRE(5,5), RRE(5,6), RRE(5,7)]`

I think it's probably supposed to return `[RRE(5,5), RRE(6,6), RRE(7,7)]`, so that we can get the list of ports we're going to schedule on.

Assuming that's right, then this would probably be less error-prone as a while loop:
```
while (begin <= end && requiredCount > 0) {
    retVal.add(new RangeResourceEntry(begin, begin));
    ++begin;
}